nexus/src/external_api/http_entrypoints.rs

```diff
@@ -3392,20 +3392,22 @@ async fn instance_serial_console(
 async fn instance_serial_console_stream_v1(
     rqctx: RequestContext<Arc<ServerContext>>,
     path_params: Path<params::InstancePath>,
-    query_params: Query<params::OptionalProjectSelector>,
+    query_params: Query<params::InstanceSerialConsoleRequest>,
```

Comment:
On `main` this is fixed by taking both sets of query params:

```rs
query_params: Query<params::InstanceSerialConsoleRequest>,
selector_params: Query<params::OptionalProjectSelector>,
```

Reply:
dropshot doesn't currently like that for this function, though - too many params.
> the trait `dropshot::handler::HttpHandlerFunc<_, _, _>` is not implemented for fn item `fn(RequestContext<Arc<context::ServerContext>>, dropshot::Path<InstancePath>, dropshot::Query<InstanceSerialConsoleRequest>, dropshot::Query<OptionalProjectSelector>, WebsocketUpgrade) -> impl futures::Future<Output = Result<http::Response<hyper::Body>, dropshot::HttpError>> {external_api::http_entrypoints::<impl From<instance_serial_console_stream> for ApiEndpoint<<RequestContext<Arc<context::ServerContext>> as RequestContextArgument>::Context>>::from::instance_serial_console_stream}`

however, for reasons that currently escape me, adding `impl_HttpHandlerFunc_for_func_with_params!((0, T1), (1, T2), (2, T3), (3, T4));` to https://github.com/oxidecomputer/dropshot/blob/main/dropshot/src/handler.rs#L357 was insufficient to resolve this.

Reply:
Weird! That will be an @ahl question I think.

Reply:
it occurs to me now that the removal of the non-v1 API call means there's no reason not to just throw it in with the rest of InstanceSerialConsoleRequest. so i'm doing that